crates/router/src/connector/bamboraapac/transformers.rs

```diff
@@ -1,234 +1,838 @@
+                        </dts:trnXML>
+                    </dts:SubmitSingleCapture>
+                </soapenv:Body>
+            </soapenv:Envelope>
+        "#,
+        receipt,
+        req.amount,
+        auth_details.username.peek(),
+        auth_details.password.peek(),
+    );
+
+    Ok(body.as_bytes().to_vec())
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename = "Envelope")]
+#[serde(rename_all = "PascalCase")]
+pub struct BamboraapacCaptureResponse {
+    body: CaptureBodyResponse,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct CaptureBodyResponse {
+    submit_single_capture_response: SubmitSingleCaptureResponse,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct SubmitSingleCaptureResponse {
+    submit_single_capture_result: SubmitSingleCaptureResult,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct SubmitSingleCaptureResult {
+    response: CaptureResponse,
 }
 
-impl<F> TryFrom<&BamboraapacRouterData<&types::RefundsRouterData<F>>> for BamboraapacRefundRequest {
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct CaptureResponse {
+    response_code: u8,
+    receipt: String,
+    declined_code: Option<String>,
+    declined_message: Option<String>,
+}
+
+impl<F>
+    TryFrom<
+        types::ResponseRouterData<
+            F,
+            BamboraapacCaptureResponse,
+            types::PaymentsCaptureData,
+            types::PaymentsResponseData,
+        >,
+    > for types::RouterData<F, types::PaymentsCaptureData, types::PaymentsResponseData>
+{
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        item: &BamboraapacRouterData<&types::RefundsRouterData<F>>,
+        item: types::ResponseRouterData<
+            F,
+            BamboraapacCaptureResponse,
+            types::PaymentsCaptureData,
+            types::PaymentsResponseData,
+        >,
     ) -> Result<Self, Self::Error> {
-        Ok(Self {
-            amount: item.amount.to_owned(),
-        })
+        let response_code = item
+            .response
+            .body
+            .submit_single_capture_response
+            .submit_single_capture_result
+            .response
+            .response_code;
+        let connector_transaction_id = item
+            .response
+            .body
+            .submit_single_capture_response
+            .submit_single_capture_result
+            .response
+            .receipt;
+        // transaction approved
+        if response_code == 0 {
+            Ok(Self {
+                status: enums::AttemptStatus::Charged,
+                response: Ok(types::PaymentsResponseData::TransactionResponse {
+                    resource_id: types::ResponseId::ConnectorTransactionId(
+                        connector_transaction_id.to_owned(),
```

Comment:
You are over-riding the authorize transaction id to capture_transaction_id, and losing the context of first_id completely. Is this intended?

Reply:
We will not be able to recover the payment_status if timeout happens in capture call